security/sshdo/Portfile

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           makefile 1.0
+
+github.setup        raforg sshdo 1.1.1 v
+revision            0
+
+categories          security
+license             GPL-2+
+maintainers         {raf.org:raf @macportsraf} openmaintainer
+description         controls which commands may be executed via incoming ssh
+long_description    {*}${description}. \
+                    \n\nsshdo provides an easily configurable way of controlling which \
+                    commands may be executed via incoming ssh connections by specific \
+                    users and specific keys. Training mode allows a key to be used for \
+                    any command, while logging them. Then sshdo can learn from the logs \
+                    to know what commands to allow. It can also unlearn commands that \
+                    are no longer in use. This mitigates against private ssh key compromise \
+                    with very little effort.
+# Any of Python 2.6, 2.7, or 3.3+ will do.
+# When updating the python3 version, update the corresponding patch below as well.
+# Update the python3 version every year (or until there's a meta port:python3).
+depends_lib-append  port:python311
+
+homepage            https://raf.org/sshdo/
+github.tarball_from releases
+checksums           rmd160 3071b9aa06452437f9da970f0576c94ea27d3c08 \
+                    sha256 9b22e14fec786b692e8d6431a81e9b17ad8f6e62ae5d6ef9e9c643690bf33f90 \
+                    size 79485
+
+# Replace "python3" with the specific version depended-on above.
+# This means the user doesn't have to select a default python3 version.
+patchfiles          macports.patch
+
+# Note: Not a GNU autoconf configure script
+use_configure       yes
+configure.args      --macports-prefix=${prefix}
```

Comment:
I'm curious why this project has a `--macports-prefix` flag. Why doesn't it use the standard `--prefix` flag that MacPorts already sets in `configure.pre_args`?

Reply:
The short answer is that the configure script isn't a standard GNU autoconf configure script. The long answer is:

It does have a --prefix argument, but that only changes the PREFIX in the Makefile which, by default, only affects where executables and manual entries are installed. It doesn't affect where configuration files are installed. That's based on the convention that I'm used to on Linux and Solaris, where the PREFIX might be e.g. /usr or /usr/local, but /etc is used for config files (i.e., not /usr/etc or /usr/local/etc). Placing /etc under the PREFIX as well seems to be more of a BSD thing.

So --prefix alone isn't suitable for macports. The configure script also has a --macports argument which does everything that's needed for macports but it assumes that the prefix is always /opt/local, which I've been informed isn't always the case. So I added --macports-prefix which is the same as --macports but with a non-default prefix supplied.

The configure script without any arguments will put config files in the right place for different systems, but I followed the Linux/Solaris convention on macOS by default, and only switch to the BSD convention with --macports or --macports-prefix.

And it's a little more complicated, because sshdo needs to read config files from under ${prefix}/etc, and one of those files needs to refer to another config file in the same directory, so it's more than just the Makefile that needs to be modified. The sshdo script itself, and the main config file also need to be changed know that /etc files are in ${prefix}/etc. If sshdo were a C program, CPPFLAGS in the Makefile could be used so that the program source itself didn't need to be modified, but it's a shell script. The configure script also changes the default source of auth.* syslog messages to /var/log/system.log (even though that's no longer used for that on new macOS systems, it is used for auth.* syslog messages on older macOS versions).

Reply:
We have over 36,000 ports in MacPorts. Not all of them use configure scripts, of course, but I can't recall any where setting just `--prefix` didn't also use that value to set the sysconfdir.

I recommend you strive to eliminate anything MacPorts specific from this software. Anything you need to do for MacPorts, you probably also need to do for some other package managers or some other subset of users, so find a generic way to make those features available in your configure script or build system. For example, if you need to use a different logfile on macOS, you can have your configure script do so automatically when building for Darwin—in fact it looks like your configure script already does that. And it looks like your configure script already supports an `--etcdir` flag, so that can be used in combination with `--prefix` without any need for a separate `--macports-prefix` option.

Reply:
We don't need to use the `--macports-prefix` option. That's just there for convenience. I've changed it to use the equivalent set of other options explicitly:

    --prefix="${prefix}" --etcdir="${prefix}"/etc --enable-mangz